viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie = {}
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    movie = next((item for item in user_data["watchlist"] if item["title"] == movie_title), None)
+    if movie:
+        add_to_watched(user_data, movie)
+        user_data["watchlist"].remove(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = 0
+    number_of_ratings = len(user_data["watched"])
+    for i in range(number_of_ratings):
+        rating_total += user_data["watched"][i]["rating"]
+    if number_of_ratings > 0:
+        return rating_total / number_of_ratings
+    else:
+        return 0

```

Comment:
This totally works in this case but I believe the README instructions said to return `0.0` when there's an empty watchlist. It seems benign and the tests still pass but it's a good practice to return the expected type (a float) in case a caller of the function is expecting that.